### ISSUE TITLE
Speedup behavior

### ISSUE THREAD

**Issue by EddyHg80** [visualboyadvance-m/visualboyadvance-m#690]
So I just updated from the latest release to the nightly version.
(Btw the release has choppy framerates, with frameskip = 0, but in the status bar it shows 60 fps regardless)

I have a 144hz screen, in the latest release I could choose 
- Speedup throttle XXX%
- Speedup frame skip 0

That way the emulator would run at 144fps speeding up the game to 240% without losing any frames. Super smooth.
If you set Speedup frame skip 1 it would double the speed. Perfect. 
A little user unfriendly because the speedup throttle became useless but really cool.

Now in the nightly there's 
- Speedup throttle XXX%
- Frame Skip checkbox

Setting 200 and box checked it goes to 200% gamespeed, but the framerate is still 60 fps. I thing it's the correct behavior.
But if I uncheck the frame skip box the game actually slows down to 43-50 fps and gamespeed tied with it to 70-80%.

So what's the deal?

**Comment by denisfa** [comment]
@EddyHg80 Are you on Windows or Linux?

> (Btw the release has choppy framerates, with frameskip = 0, but in the status bar it shows 60 fps regardless)

Does the nightly still have this issue?

> I have a 144hz screen, in the latest release I could choose

Notebook or desktop? What graphic card? There are some edge cases here, hence why I need this info.

> A little user unfriendly because the speedup throttle became useless but really cool.

They are alternatives. If one does not work, you can try the other. If they stacked together, it would be even more confusing.

> Setting 200 and box checked it goes to 200% gamespeed, but the framerate is still 60 fps. I thing it's the correct behavior.

This is correct indeed.

> But if I uncheck the frame skip box the game actually slows down to 43-50 fps and gamespeed tied with it to 70-80%.
> So what's the deal?

Could be a couple of things... Would you upload your `vbam.ini`, please?

**Comment by EddyHg80** [comment]
> @EddyHg80 Are you on Windows or Linux?
> 
> > (Btw the release has choppy framerates, with frameskip = 0, but in the status bar it shows 60 fps regardless)
> 
> Does the nightly still have this issue?

Nope, the nightly seems to be smooth on normal speed, that's why I updated.

> 
> > I have a 144hz screen, in the latest release I could choose
> 
> Notebook or desktop? What graphic card? There are some edge cases here, hence why I need this info.

It's a notebook, nvidia gtx 1650 mobile.

> 
> > A little user unfriendly because the speedup throttle became useless but really cool.
> 
> They are alternatives. If one does not work, you can try the other. If they stacked together, it would be even more confusing.
> 
> > Setting 200 and box checked it goes to 200% gamespeed, but the framerate is still 60 fps. I thing it's the correct behavior.
> 
> This is correct indeed.
> 
> > But if I uncheck the frame skip box the game actually slows down to 43-50 fps and gamespeed tied with it to 70-80%.
> > So what's the deal?
> 
> Could be a couple of things... Would you upload your `vbam.ini`, please?

Sure!

[vbam.zip](https://github.com/visualboyadvance-m/visualboyadvance-m/files/4813910/vbam.zip)

I retested now, with frame skip unchecked I don't get the slowdown anymore, I get 61-62 fps, and the audio is very distorted.

Still I would love the previous behavior: no frameskip so 144fps 240% speed, and maybe when you check frameskip it goes as fast as it can with a reasonable frame skip value, but still 144fps (or any hz you have on the screen).

**Comment by denisfa** [comment]
@EddyHg80 Thank you for the extra. Let's see what we can do.

> I retested now, with frame skip unchecked I don't get the slowdown anymore, I get 61-62 fps, and the audio is very distorted.

I believe you are using the integrated graphic card to run vbam. Your nvidia graphic card is the one for `144Hz`, the integrated one is probably `60Hz`. Could you please do a couple of tests for me? Make sure `vsync` is not checked on menu (you need to do this before loading a game to take effect).

Set/Replace the following options on your `vbam.ini`:
```
speedupThrottle=240
speedupThrottleFrameSkip=0
vsync=0
```

Then, run `vbam` selecting the integrated graphic card and then using the nvidia card. Do you still have the same issues?

**Comment by EddyHg80** [comment]
Ok so I can't actually choose which graphics card to use, I think my cpu i7-9750HF hasn't an integrated graphincs card.
I have the nvidia thingy in the bottom right corner and I confirm I'm running the emulator with the 1650.

With those settings I get 235-240% game speed but 45-47 fps, clearly with so much frameskip.

**Comment by denisfa** [comment]
@EddyHg80 Alright, thank you for the feedback. I will investigate the code now.

**Comment by rkitover** [comment]
> With those settings I get 235-240% game speed but 45-47 fps, clearly with so much frameskip.

Game speed is controlled by the sound driver, can you try different sound drivers and see if you get the same results?

Which sound driver are you using right now?

**Comment by EddyHg80** [comment]
How can I change which sound driver the emulator is using? I think I have Realtek audio

**Comment by rkitover** [comment]
I meant our own sound drivers.

Go to `Options -> Audio -> Configure -> Advanced`.

**Comment by EddyHg80** [comment]
I actually tried all of them after your reply because I wasn't sure. Same behavior on all of them.
I was on SDL.
The other 3 have even worst performance.

**Comment by rkitover** [comment]
Great, thank you!

**Comment by rkitover** [comment]
@denisfa this one might be difficult to track down, I suggest doing a bisect, see here:

https://www.metaltoad.com/blog/beginners-guide-git-bisect-process-elimination
